File: omc/android/app/src/main/java/com/pureenee/model/SanamentoDebito.java
package com.pureenee.model;

import io.objectbox.annotation.Entity;
import io.objectbox.annotation.Id;
import io.objectbox.relation.ToOne;

@Entity
public class SanamentoDebito {

    @Id
    long id;

    double importoSanamento;

    long dataSanamento;

    ToOne<Debt> debito;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public double getImportoSanamento() {
        return importoSanamento;
    }

    public void setImportoSanamento(double importoSanamento) {
        this.importoSanamento = importoSanamento;
    }

    public long getDataSanamento() {
        return dataSanamento;
    }

    public void setDataSanamento(long dataSanamento) {
        this.dataSanamento = dataSanamento;
    }

    public Debt getDebito() {
        return debito.getTarget();
    }

    public void setDebito(Debt debito) {
        this.debito.setTarget(debito);
    }
}

File: omc/lib/model/sanamento_debito.dart

import 'debt.dart';

class SanamentoDebito {

  int? id;

  double importoSanamento;

  int dataSanamento;

  Debt? debito;

  SanamentoDebito(
      this.id,
      this.importoSanamento,
      this.dataSanamento,
      this.debito
  );

  factory SanamentoDebito.fromJson(Map<String, dynamic> json) {
    var debito = null;

    if(json['debito'] != null){
       debito = Debt.fromJson(json['debito']);
    }


    return SanamentoDebito(
        json['id'],
        json['importoSanamento'],
        json['dataSanamento'],
        debito
    );
  }

  Map<String, dynamic> toJson() => {
    'id':id,
    'importoSanamento': importoSanamento,
    'dataSanamento':dataSanamento,
    'debt': debito
  };

}